lib/remap-lang/src/expression/arithmetic.rs

```diff
@@ -0,0 +1,48 @@
+use super::{Expr, Expression, Object, Result, State, Value};
+use crate::Operator;
+
+#[derive(Debug)]
+pub(crate) struct Arithmetic {
```

Comment:
Consider prefixing this to distinguish it from unary arithmetic (`-` negate, `~` binary invert, and `!` logical invert).

Reply:
The alternative would be to make this an enum using a strategy similar to enum dispatch above:
```rust
enum Arithmetic {
    Multiply(Box<Expr>, Box<Expr>),
    Negate(Box<Expr>),
}
```
That's probably more work now.

Reply:
This is mostly moved over from the existing impl. We don't support negate and binary invert arithmetic, and logical/not has its own expression implementation.

I'm going to leave this as is for now, but we can iterate on this if/when we do add all unary arithmetic options.